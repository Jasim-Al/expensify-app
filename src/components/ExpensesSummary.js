import React from 'react';
import { connect } from 'react-redux';
import expensesSelector from '../selectors/expenses';
import expensesSum from '../selectors/expenses-total';
import numeral from 'numeral'

export const ExpensesSummary = props => (
    <div>
       <h1>Viewing {props.expensesLength} {props.expensesLength > 1 ? 'expenses' : 'expense' } totalling {numeral(props.sum/100).format('$0,0.00')} </h1>
    </div>
);

const mapStateToProps = state => {
    return {
        expensesLength : expensesSelector(state.expenses, state.filters).length,
        sum:expensesSum(state.expenses)
    };
};

export default connect(mapStateToProps)(ExpensesSummary);