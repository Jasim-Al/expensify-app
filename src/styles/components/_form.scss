.form {
    display: flex;
    flex-direction: column;
    div{
        div{
            width: 100%;
        }
    }
    > * {
        margin-bottom: $m-size;
    }
}

.form__error {
    margin: 0 0 $m-size 0;
    font-style: italic;
}