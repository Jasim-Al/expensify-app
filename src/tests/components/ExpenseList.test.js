import React from 'react';
import {shallow} from 'enzyme';
import {ExpenseList} from '../../components/ExpenseList';
import expenses from '../fixtures/expenses';

test('render ExpenseList with expenses', () => {
    let wrapper = shallow( <ExpenseList expenses = { expenses } /> );
    expect(wrapper).toMatchSnapshot();
});

test('render ExpenseList with with empty expenses message', () => {
    let wrapper = shallow( <ExpenseList expenses = { [] } /> );
    expect(wrapper).toMatchSnapshot();
});