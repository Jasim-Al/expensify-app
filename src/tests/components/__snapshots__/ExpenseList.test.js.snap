// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render ExpenseList with expenses 1`] = `
<div>
  <ExpenseListItems
    amount={100}
    createdAt={0}
    description="shawarma"
    id="1"
    key="1"
    note=""
  />
  <ExpenseListItems
    amount={70}
    createdAt={-345600000}
    description="burger"
    id="2"
    key="2"
    note=""
  />
  <ExpenseListItems
    amount={120}
    createdAt={345600000}
    description="pizza"
    id="3"
    key="3"
    note=""
  />
</div>
`;

exports[`render ExpenseList with with empty expenses message 1`] = `
<div>
  <p>
    No Expenses
  </p>
</div>
`;
